src/main/java/seedu/condonery/logic/parser/AddCommandParser.java

```diff
@@ -18,29 +19,49 @@
 /**
  * Parses input arguments and creates a new AddCommand object
  */
-public class AddCommandParser implements Parser<AddCommand> {
+public class AddCommandParser implements Parser<Command> {
+    private final String directoryArgument;
+
+    /**
+     * Initialises an AddCommandParser with the given directoryArgument.
+     */
+    public AddCommandParser(String directoryArgument) {
```

Comment:
I think this works! Wondering if we should create an abstract PropertyCommand class since all subclasses should follow this behaviour of accepting a "-p". 

Also think AddPropertyCommand might be a clearer class name. WDYT?

Reply:
@zacchaeuschok

Reply:
Oops, I agree with you that creating abstract PropertyCommand and ClientCommand might be a bit verbose. 

But in terms of inheritance, would it make sense for there to be PropertyCondoneryParser and ClientCondonerParser that inherit from CondoneryParser classes? The difference would be the BASIC_COMMAND_FORMAT pattern.

Reply:
Oh, I think that's a good suggestion. Am I correct in thinking that you're thinking of ways to split the Property and Client logic so the code doesn't look so messy?

After reviewing the related classes, I think the only way to properly split the logic is if we create a new set of either `CommandParser` classes (`AddPropertyCommandParser`, `AddClientCommandParser`, `DeletePropertyCommandParser` etc.) or `Command` classes (`AddPropertyCommand`, `AddClientCommand` etc.). To reduce the mess, we could package them into logic.commands.property and logic.commands.client. WDYT?

Reply:
Yup, I think creating `CommandParser` classes (`AddPropertyCommandParser` and `AddClientCommandParser`) is a good start. 

Then, identifying the `-p` and `-c` flag can be done in the `CondoneryParser` class.

Reply:
Ok, thanks for your input! I'll work on it and will push to the PR once done 👍

Reply:
Should we also have an interface, since adding Property and adding Clients should have similar methods

Reply:
That might be a good idea! I think maybe we can finish up the implementation for all the commands first then look at further ways to make it neater/follow the OOP principles better

Reply:
sounds good to me! I agree with Zac that having Property/Client CondoneryParser would be good too. I think to follow the existing codebase we probably have to add a Parser + Command for every command we add, unfortunately.